### PR TITLE
docs: add guide, refactor doc structure

### PR DESCRIPTION
**Pull request by andrewg-xyz**

### CODE REVIEW

**Comment by andrewg-xyz** [comment]
I copied work from https://github.com/defenseunicorns/uds-marketplace/pull/107 as uds-common is a more appropriate location.

Also refactored the docs a bit to introduce some structure.

File: docs/package_integration/guide.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+## Prerequisites
+
+- UDS generally assumes applications are containerized and can be deployed via Kubernetes. (exceptions are possible, but rare, and require additional work e.g. KubeVirt)
+- The integrator (you) should have access to the application image and deployment code.
+- The "App" should have a compatible license for UDS integration (Apache 2.0 or similar) [see Licensing Considerations](#licensing-considerations).
+
+### Licensing Considerations
+
+- Defense Unicorns uses Apache 2.0 licensed products exclusively, see the [Open Source Policy](https://github.com/defenseunicorns/uds-common/blob/main/docs/adrs/0002-apache-2.0-for-all-uds-products.md).
```
Comment (Racer159) [inline docs/package_integration/guide.md:21]:
```suggestion
- Defense Unicorns creates Apache 2.0 licensed products exclusively, see the [Open Source Policy](https://github.com/defenseunicorns/uds-common/blob/main/docs/adrs/0002-apache-2.0-for-all-uds-products.md).
```
We use AGPL/MPL products inside some packages - just our own stuff is only Apache 2

File: docs/package_integration/guide.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+When integrating an application with UDS, consider the following requirements and suggestions:
+
+### Step 1 - Zarf Package
+
+Your goal is to bundle the upstream helm chart and images into a single Zarf package, defined in a `zarf.yaml`
+
+*reminder https://docs.zarf.dev*
+
+- [ ] Understand zarf's [helm chart component](https://docs.zarf.dev/ref/components/#helm-charts)
```
Comment (Racer159) [inline docs/package_integration/guide.md:46]:
```suggestion
- [ ] Understand Zarf's [Helm Chart component primitive](https://docs.zarf.dev/ref/components/#helm-charts)
```

File: docs/package_integration/guide.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+- [ ] The Command `uds zarf dev generate` may be useful to generate an initial `zarf.yaml` file for your application.
+- [ ] Identify the application's images, the command `uds zarf dev find-images` may be useful.
+- [ ] The application may require additional configurations at build time or runtime, consider [Zarf Component Actions](https://docs.zarf.dev/ref/examples/component-actions/)
+- [ ] Start to monitor for the application reaching out to the internet For example [sonarqube](https://github.com/defenseunicorns/uds-package-sonarqube/tree/main/src/monitoring-image) Sonarqube tries to download a jar file from Maven central to enable monitoring with Prometheus (it is not baked into their images) - this custom image takes a Chainguard busybox base, adds the correct jar file, and creates a simple script that calls itself "curl" that doesn't curl and instead copies the Jar file it already has to make that piece work in the Air Gap. Its basically a shim around the way the upstream chart works.
```
Comment (Racer159) [inline docs/package_integration/guide.md:50]:
```suggestion
- [ ] Start to monitor for the application reaching out to the internet For example [Sonarqube](https://github.com/defenseunicorns/uds-package-sonarqube/tree/main/src/monitoring-image) tries to download a jar file from Maven central to enable monitoring with Prometheus (it is not baked into their images) - this custom image takes a Chainguard busybox base, adds the correct jar file, and creates a simple script that calls itself "curl" that doesn't curl and instead copies the Jar file it already has to make that piece work in the Air Gap. Its basically a shim around the way the upstream chart works.
```

File: docs/package_integration/guide.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+#### Checkout
+Your repository has a `zarf.yaml`, you can build a single artifact with `uds zarf package create`, and deploy it to a [k3d-core-dev-slim cluster](https://github.com/defenseunicorns/uds-core?tab=readme-ov-file#uds-package-development) using `uds zarf package deploy`
+
+### Step 2 - UDS Package
+
+Your goal is to integrate the Zarf package application with [UDS Core](https://github.com/defenseunicorns/uds-core), using `uds-package.yaml` custom resource.
+
+- [ ] If you haven't already, read the docs on [UDS Operator](https://uds.defenseunicorns.com/core/configuration/uds-operator/)
+
+- [ ] [UDS Package Practices](https://github.com/defenseunicorns/uds-common/blob/main/docs/uds-package-practices.md) provides an extensive list of best practices, considerations, and tasks to for package
```
Comment (Racer159) [inline docs/package_integration/guide.md:61]:
```suggestion
- [ ] [UDS Package Practices](https://github.com/defenseunicorns/uds-common/blob/main/docs/uds-package-practices.md) provides an extensive list of best practices, considerations, and tasks for a package to be "Made for UDS"
```

File: docs/package_integration/uds-package-practices.md [hunk 5]
```diff
@@ -1,15 +1,26 @@
-# UDS Package Practices
+# UDS Package Integration Guide
 
-This document describes the practices that a UDS package **must**, **should** and **may** follow to be considered officially supported as a UDS package and to be able to display the "Made for UDS" badge on its GitHub repository README.md file :
+## Introduction
 
-[<img alt="Made for UDS" src="./made-for-uds.svg" height="20px"/>](https://github.com/defenseunicorns/uds-core)
+This guide is intended for developers integrating applications with UDS (Unicorn Delivery Service). This document describes the practices that a UDS package _**must**_, _**should**_ and _**may**_ follow to be considered officially supported as a UDS package.
```
Comment (Racer159) [inline docs/package_integration/uds-package-practices.md:5]: do we need these edits here? isn't the other file the guide?

**Review by Racer159 — CHANGES_REQUESTED** [review]

File: docs/package_integration/uds-package-practices.md (hunk 5 above)
Comment (andrewg-xyz) [reply docs/package_integration/uds-package-practices.md:5]: Good catch, My copy/paste fell short. will update

File: docs/package_integration/guide.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+#### Checkout
+Your repository has a `uds-package.yaml` manifest added to the appropriate helm chart and you can deploy to a K3d Core Dev Slim cluster, via zarf package.
+
+## Examples
+
+For reference, consider these well-maintained UDS package examples:
+- [UDS Package GitLab](https://github.com/defenseunicorns/uds-package-gitlab) (More complex example)
+- [UDS Package Mattermost](https://github.com/defenseunicorns/uds-package-mattermost) (Simpler example)
+- [UDS Common NGINX](https://github.com/defenseunicorns/uds-common/pull/191)
```
Comment (Racer159) [inline docs/package_integration/guide.md:71]:
```suggestion
- [UDS Common NGINX](https://github.com/defenseunicorns/uds-common)
```
Didn't check the links close enough.

File: docs/package_integration/uds-package-practices.md [hunk 7]
```diff
@@ -1,15 +1,26 @@
-# UDS Package Practices
+# UDS Package Integration Guide
```
Comment (Racer159) [inline docs/package_integration/uds-package-practices.md:1]: would prefer to have one "guide" to reduce confusion on naming (also would argue this docs  was meant more to be a summary of requirements than a guide on how to integrate with UDS - that was more what I meant than specific wording on that line)

**Review by Racer159 — CHANGES_REQUESTED** [review]

**Review by Racer159 — DISMISSED** [review]

**Review by Racer159 — APPROVED** [review]

**Pull request merged**